io-engine/src/subsys/nvmf/subsystem.rs

```diff
@@ -165,6 +194,169 @@ fn make_sn<T: AsRef<[u8]>>(uuid: T) -> CString {
 }
 
 impl NvmfSubsystem {
+    fn reset_cb(success: bool, ctx: *mut c_void) {
+        if success {
+            info!(
+                "nvmf_event_handler: controller successfully reset {:?}",
+                success
+            );
+        } else {
+            error!("nvmf_event_handler: failed to reset controller ");
+        }
+        done_cb(ctx, success);
+    }
+
+    pub async fn reset_one_child(dev_name: String, ctx: *mut c_void) {
+        let ctrlr = match NVME_CONTROLLERS.lookup_by_name(&dev_name) {
+            Some(ctrlr) => ctrlr,
+            None => {
+                debug!("device name  {dev_name} not a valid NVMe controller");
+                Self::reset_cb(false, ctx);
+                return;
+            }
+        };
+        debug!("nvmf_event_handler: nvme controller {:?} ", ctrlr);
+        let mut ctrlr = ctrlr.lock();
+        if ctrlr.reset(NvmfSubsystem::reset_cb, ctx, false).is_ok() {
+            debug!("nvmf_event_handler: Reset initiated for {:?} ", ctrlr);
+        } else {
+            error!(
+                "nvmf_event_handler: Failed to initiate reset for {:?} ",
+                ctrlr
+            );
+            Self::reset_cb(false, ctx);
+        }
+    }
+
+    pub async fn reset_nexus(nexus_name: String) {
+        let mut nexus = match nexus_lookup_mut(&nexus_name) {
+            Some(v) => v,
+            None => {
+                debug!("nqn {nexus_name} does not contain a valid nexus");
+                return;
+            }
+        };
+
+        //Pause the Nexus
+        if nexus.as_mut().pause().await.is_err() {
+            error!("Failed to pause Nexus {:?}", nexus);
+            return;
+        }
+        info!("Paused the nexus - {:?} ", nexus_name);
+        for child in nexus.children_iter() {
+            debug!("Found a {:?} for nexus {nexus_name}", child);
+            if !child.is_healthy() {
+                continue;
+            }
+            let dev = match child.get_device() {
+                Ok(dev) => dev,
+                Err(error) => {
+                    error!(
+                        "Nexus {nexus_name} get_device failed for {:?} with Error {:?}",
+                         child, error
+                    );
+                    continue;
+                }
+            };
+            let dev_name = dev.device_name();
+            if dev.driver_name() != "nvme" {
+                debug!("Device {dev_name} is not NVMe-oF, reset is not needed");
```

Comment:
Why is reset not needed for local replicas?

Reply:
The whole issue related to NVMeOF protocol, the replica side responding to some old commands after the initiator restarted.

Reply:
Is there a test to confirm this PR fixes that issue?